starknet_py/net/client_test.py

```diff
@@ -74,3 +94,145 @@ def test_get_rpc_storage_key(key, expected):
 def test_get_rpc_storage_key_raises_on_non_representable_key(key):
     with pytest.raises(ValueError, match="cannot be represented"):
         _to_storage_key(key)
+
+
+@pytest.mark.asyncio
+async def test_broadcasted_txn_declare_v3(
+    account, abi_types_compiled_contract_and_class_hash
+):
+    declare_v3 = await account.sign_declare_v3(
+        compiled_contract=abi_types_compiled_contract_and_class_hash[0],
+        compiled_class_hash=abi_types_compiled_contract_and_class_hash[1],
+        l1_resource_bounds=MAX_RESOURCE_BOUNDS_L1,
+    )
+
+    prev_brodcasted = _create_broadcasted_txn_prev(declare_v3)
+    brodcasted = _create_broadcasted_txn(declare_v3)
+    ddiff = DeepDiff(prev_brodcasted, brodcasted, ignore_order=True)
+
+    assert brodcasted["type"] == TransactionType.DECLARE.name
+    del brodcasted["type"]
+    DeclareBroadcastedV3Schema(exclude=["class_hash"]).load(brodcasted)
```

Comment:
Could you clarify what exactly we are testing with these lines?

Reply:
We check if `brodcasted` fit to the schema defined in `DeclareBroadcastedV3Schema`. Marshmallow.dump don't verify if all required files are contained in dict

Reply:
I believe there may be ways to improve these tests. For instance, one suggestion I have is verifying if the broadcasted transaction incorporates all the expected keys. This can look more or less like this:
```
brodcasted_txn = _create_broadcasted_txn(invoke_v1)
expected_keys = dataclasses.fields(InvokeV1) # or dataclasses.fields(invoke_v1)
assert all(key.name in brodcasted_txn for key in expected_keys)
```